plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/ray_aws_launcher.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+import os
+import tempfile
+from pathlib import Path
+from subprocess import PIPE, Popen
+from typing import Any, Dict, List, Optional, Sequence, Tuple
+
+import ray.cloudpickle as cloudpickle  # type: ignore
+from omegaconf import DictConfig, OmegaConf, open_dict
+
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.singleton import Singleton
+from hydra.core.utils import JobReturn, configure_log, filter_overrides, setup_globals
+from hydra.plugins.launcher import Launcher
+from hydra.types import TaskFunction
+from hydra_plugins.hydra_ray_launcher.ray_launcher_util import (
+    JOB_RETURN_PICKLE,
+    JOB_SPEC_PICKLE,
+)
+
+log = logging.getLogger(__name__)
+
+
+class RayAWSLauncher(Launcher):
+    def __init__(
+        self,
+        ray_init_cfg: DictConfig,
+        ray_remote_cfg: DictConfig,
+        ray_cluster_cfg: DictConfig,
+        stop_cluster: bool,
+    ) -> None:
+        self.ray_init_cfg = ray_init_cfg
+        self.ray_remote_cfg = ray_remote_cfg
+        self.ray_cluster_cfg = ray_cluster_cfg
+        self.ray_yaml_path = self._save_cluster_config()
+        self.docker_enabled = (
+            OmegaConf.select(self.ray_cluster_cfg, "docker.image") != ""
+        )
```

Comment:
now that you have a guarantee from the structured config that this exists you can probably simplify this.